rest-server/core/src/main/scala/com/actionml/core/backup/FSMirroring.scala

```diff
@@ -18,70 +18,67 @@
 
 import java.io.{File, FileWriter, PrintWriter}
-import java.nio.file.{Files, Path, Paths}
 
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
 import com.actionml.core.template.Engine
+import com.actionml.core.validate.{ValidRequestExecutionError, ValidateError}
 
 import scala.io.Source
 
 /**
   * Mirroring implementation for local FS.
   */
 
-// TODO: not using injection so need a trait
-//object FSMirroring extends Mirroring {
-trait FSMirroring extends Mirroring {
+class FSMirroring(mirrorContainer: String) extends Mirroring(mirrorContainer) {
 
   val f = new File(mirrorContainer)
-  if( f.exists() && f.isDirectory() && config.getString("mirror.type").nonEmpty) {
-    logger.info(s"Morroring raw un-validated events to ${mirrorContainer}")
-  } else if(config.getString("mirror.type").nonEmpty){
-    logger.error(s"Mirror location: ${mirrorContainer} not configured to accept event mirroring.")
+  if (f.exists() && f.isDirectory && config.getString("mirror.type").nonEmpty) {
+    logger.info(s"Morroring raw un-validated events to ${ mirrorContainer }")
+  } else if (config.getString("mirror.type").nonEmpty) {
+    logger.error(s"Mirror location: ${ mirrorContainer } not configured to accept event mirroring.")
   } else {
     logger.warn(s"Mirroring location is set but type is not configured, no mirroring with be done.")
   }
 
 
-
   // java.io.IOException could be thrown here in case of system errors
-  override def mirrorEvent(engineId: String, json: String): Boolean = {
-    var noError = true
-    if(mirrorType == Mirroring.localfs){
+  override def mirrorEvent(engineId: String, json: String): Validated[ValidateError, Unit] = {
+    def mirrorEventError(errMsg: String) =
+      Invalid(ValidRequestExecutionError(s"Unable to mirror event: $errMsg"))
+
       try {
         val resourceCollection = new File(containerName(engineId))
         //logger.info(s"${containerName(engineId)} exists: ${resourceCollection.exists()}")
-        if( !resourceCollection.exists()) new File(s"${containerName(engineId)}").mkdir()
-        val pw = new PrintWriter(new FileWriter(s"${containerName(engineId)}/$batchName.json", true))
+        if (!resourceCollection.exists()) new File(s"${ containerName(engineId) }").mkdir()
+        val pw = new PrintWriter(new FileWriter(s"${ containerName(engineId) }/$batchName.json", true))
         try {
           pw.write(json)
         } finally {
           pw.close()
         }
       } catch {
-        case e: Exception =>
-          logger.error("Problem mirroring while input")
-          e.printStackTrace
-          noError = false
+        case ex: Exception =>
+          val errMsg = "Problem mirroring while input"
+          logger.error(errMsg, ex)
+          mirrorEventError(s"$errMsg: ${ex.getMessage}")
       }
-
-    } else {
-      logger.warn("Local filesystem mirroring called, but not configured.")
-      noError = false
-    }
-    noError
+    Valid(())
   }
```

Comment:
We have been using Validated only when input is parsed or external commands are sent in, not for general error return values. Is this used later building the response to input?